### PR TITLE
[Tests] Remove all hard-coded macOS targets in explicit module tests

### PR DESCRIPTION
**Pull request by cachemeifyoucan**
Remove all the hard coded macOS tests for module scanning and planning. Feed stdlib and shims into the tests instead to support all targets for those tests.

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Review by compnerd — APPROVED** [review]

**Comment by cachemeifyoucan** [comment]
Holding on. Bots are actually not green for Linux as the test failures are hidden.

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

File: TestInputs/ExplicitModuleBuilds/CHeaders/G.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#if __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 110000
+#if defined(__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__) && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1050
```
Comment (compnerd) [inline TestInputs/ExplicitModuleBuilds/CHeaders/G.h:1]:
```suggestion
#if __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__-0 < 1050
```

**Review by compnerd — APPROVED** [review]

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

File: TestInputs/ExplicitModuleBuilds/CHeaders/G.h (hunk 1 above)
Comment (cachemeifyoucan) [reply TestInputs/ExplicitModuleBuilds/CHeaders/G.h]: I am actually intended to have this to be false on Linux/Windows.

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Comment by cachemeifyoucan** [comment]
@swift-ci please test windows platform

**Review by artemcm — APPROVED** [review]

**Pull request merged**